clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -987,29 +1039,35 @@ class CIRCmpOpLowering : public mlir::OpConversionPattern<mlir::cir::CmpOp> {
   }
 };
 
-class CIRBrOpLowering : public mlir::OpRewritePattern<mlir::cir::BrOp> {
+class CIRBrOpLowering : public mlir::OpConversionPattern<mlir::cir::BrOp> {
 public:
-  using OpRewritePattern<mlir::cir::BrOp>::OpRewritePattern;
+  using OpConversionPattern<mlir::cir::BrOp>::OpConversionPattern;
 
   mlir::LogicalResult
-  matchAndRewrite(mlir::cir::BrOp op,
-                  mlir::PatternRewriter &rewriter) const override {
-    rewriter.replaceOpWithNewOp<mlir::LLVM::BrOp>(op, op.getDestOperands(),
-                                                  op.getDest());
+  matchAndRewrite(mlir::cir::BrOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    if (op.getDestOperands().size() != 0) {
+      auto convertOp = op.getDestOperands().front().getUsers().begin();
+      rewriter.replaceOpWithNewOp<mlir::LLVM::BrOp>(op, (**convertOp).getOpResults(),
+                                                    op.getDest());
+    } else {
+      rewriter.replaceOpWithNewOp<mlir::LLVM::BrOp>(op, op.getDestOperands(),
+                                                    op.getDest());
+    }
```

Comment:
@redbopo I would suggest iterating and converting all `destOperand` types to generalize the solution for any amount of operands. This approach seems to work only for zero or one operand.

Reply:
Will fix in next step.

Reply:
Can you please add the fix to #98 then?